conda_build/build.py

```diff
@@ -582,7 +582,7 @@ def test(m, verbose=True, channel_urls=(), override_channels=False):
         else:
             test_file = join(tmp_dir, 'run_test.sh')
             # TODO: Run the test/commands here instead of in run_test.py
-            cmd = ['/bin/bash', '-x', '-e', test_file]
+            cmd = ['/bin/sh', '-x', '-e', test_file]
```

Comment:
Let me put my two cents here.

I would support using `sh` instead of `bash` if this breaking change will be made with a deprecation step (for at the very least a couple of minor releases) instead of breaking things altogether. There are some quite complex scripts already in conda-recipes repo using features of bash, for example GCC: https://github.com/conda/conda-recipes/blob/master/gcc-4.8/post-link.sh